types/react/index.d.ts

```diff
@@ -558,7 +629,41 @@ declare namespace React {
 
     function createRef<T>(): RefObject<T>;
 
-    function forwardRef<T, P = {}>(Component: RefForwardingComponent<T, P>): ComponentType<P & ClassAttributes<T>>;
+    interface ForwardRefExoticComponent<P> extends ExoticComponent<P> {
+        defaultProps?: Partial<P>;
+    }
+
+    function forwardRef<T, P = {}>(Component: RefForwardingComponent<T, P>): ForwardRefExoticComponent<P & ClassAttributes<T>>;
+
+    type ComponentProps<T extends ComponentType<any>> =
```

Comment:
Yeah, if there is no argument it'll be inferred as `{}` and this will actually cause the compiler to reject subtypes of `ComponentType<{}>` because `{}` will be a function argument.

It's the problem of `(arg: { foo: string }) => null` not being assignable to `(arg: {}) => null`, but being assignable to `(arg: any) => null`.